addons/l10n_it_edi_ndd/models/account_move.py

```diff
@@ -43,8 +47,11 @@ def _compute_l10n_it_payment_method(self):
     def _compute_l10n_it_document_type(self):
         document_type = self.env['l10n_it.document.type'].search([]).grouped('code')
         for move in self:
+            if move.country_code != 'IT' or move.l10n_it_document_type:
```

Comment:
```suggestion
            if move.country_code != 'IT' or move.l10n_it_document_type or move.state != 'posted':
```
Couldn't we group all the conditions here and get rid of the `else` statement?